be/src/exec/pipeline/scan_operator.cpp

```diff
@@ -254,6 +258,41 @@ Status ScanOperator::_pickup_morsel(RuntimeState* state, int chunk_source_index)
     return Status::OK();
 }
 
+void ScanOperator::_merge_chunk_source_profiles() {
+    std::vector<RuntimeProfile*> profiles(_chunk_source_profiles.size());
+    for (auto i = 0; i < _chunk_source_profiles.size(); i++) {
+        profiles[i] = _chunk_source_profiles[i].get();
+    }
+    RuntimeProfile::merge_isomorphic_profiles(profiles);
+
+    RuntimeProfile* merged_profile = profiles[0];
+
+    // Copy info strings
+    std::map<std::string, std::string> info_strings;
+    merged_profile->get_all_info_strings(&info_strings);
+    for (auto& [key, value] : info_strings) {
+        _unique_metrics->add_info_string(key, value);
+    }
+
+    // Copy counters
+    std::map<std::string, RuntimeProfile::Counter*> counters;
+    std::map<std::string, std::set<std::string>> child_counter_map;
+    merged_profile->get_all_counters(&counters, &child_counter_map);
+    std::queue<std::string> parent_queue;
+    parent_queue.push(RuntimeProfile::ROOT_COUNTER);
+    while (!parent_queue.empty()) {
+        std::string parent_counter_name = parent_queue.front();
```

Comment:
Maybe we could use `std::move(parent_queue.front())` to invoke move constructor instead of copy constructor for `std::string`.

Reply:
Ok